### DISCUSSION TITLE
Socket connection closes by itself when app goes in background

### DISCUSSION THREAD

**Discussion by bejeri4** [HaishinKit/HaishinKit.swift#1022] [Q&A]
I would like to keep the connection alive when app goes in background and continue audio streaming.
However `rtmpStatus` changes to `RTMPConnection.Code.connectClosed` and I have to reconnect. After reconnecting it's working - audio data is streamed as expected.
The problem is that reconnecting takes some time (up to 15 seconds) and I don't want to pause the stream.

Is there any way to avoid socket status changes when app goes in background?

 - Device: `iPhone 13`
 - OS: `iOS 15.5`
 - Version `1.2.4`

**Comment by shogo4405** [comment]
1. Please turn on Background Modes, Audio AirPlay, and Picture In Picture. <img width="666" alt="スクリーンショット 2022-07-02 10 39 16" src="https://user-images.githubusercontent.com/810189/176981976-cff880ca-4459-40e8-aa46-2a5e57863fec.png">
2. Please call `rtmpStream.attachCamera(nil)` when Apps enter background.
3. Please call `rtmpStream.attachCamera(currentPosition)` when Apps enter foreground.
   - seealso code: https://github.com/shogo4405/HaishinKit.swift/commit/265d8eea3cd28617a4a14d32ce995d8a7ef67370

**Comment by bejeri4** [comment]
Hey @shogo4405, thanks for the replay!

`Background Modes, Audio Airplay, and Picture in Picture` is turned on:

<img width="340" alt="Screenshot 2022-07-02 at 09 08 20" src="https://user-images.githubusercontent.com/29302676/176987933-cf09f719-ee62-443d-9606-fd169c269496.png">

I also added `rtmpStream.attachCamera(nil)` when app enters in background. The result is still the same.

Once again, background streaming works - when I call reconnect in background, it reconnects successfully and streams audio without any loss. The problem is that connection closes by itself when app goes in background and requires reconnection.

**Comment by bejeri4** [comment]
Also it's working on a simulator without any problems.

**Comment by bejeri4** [comment]
Can someone please help me on this? I created an issue but the repo maintainer deleted it and opened this thread instead 🥴

**Comment by aaronn** [comment]
Same issue here actually – it works on Simulator but not on an actual device. Any ideas?
